docs/server-side/v2/how--to-guides/nodejs-server-side-sdk.mdx

```diff
@@ -0,0 +1,226 @@
+const roomUpdateOptions = { name };
+const updatedRoom = await hms.rooms.update(room.id, roomUpdateOptions);
+console.log(room, roomWithOptions, updatedRoom);
+```
+
+### > Usage with TypeScript
+
+The SDK supports `ts`, `esm` and `cjs` completely. Here's how you can import the types from the SDK and use them.
+
+```ts
+import HMS from "@100mslive/server-sdk";
+
+const hms = new HMS.SDK();
+// create a room with options -
+let roomWithOptions: HMS.Room;
+const roomCreateOptions: HMS.RoomCreateOptions = {
+  name,
+  description,
+  template_id,
+  recording_info,
+  region,
+};
+roomWithOptions = await hms.rooms.create(roomCreateOptions);
+```
+
+### Create room codes for a room and then disable one of them
+
+```js
+import HMS from "@100mslive/server-sdk";
+
+const hms = new HMS.SDK();
+// create room codes -
+const roomCodesForRoom = await hms.roomCodes.create(roomId);
+console.log(roomCodesForRoom);
+
+// disable a room code -
+const disabledRoomCode = await hms.roomCodes.enableOrDisable(roomCodesForRoom[0].code, false);
+console.log(disabledRoomCode);
+```
+
+### List Peers in an active room and send a message to the room
+
+```js
+import HMS from "@100mslive/server-sdk";
+
+const hms = new HMS.SDK();
+// list peers in active room -
+const peers = await hms.activeRooms.retrieveActivePeers(roomId);
+console.log(peers);
+
+// send broadcast message to all peers -
+await hms.activeRooms.sendMessage(roomId, { message: "test" });
+```
+
+### List all sessions and the sessions in a room
+
+```js
+import HMS from "@100mslive/server-sdk";
+
+const hms = new HMS.SDK();
+// list all sessions -
+const allSessionsIterable = hms.sessions.list();
+for await (const session of allSessionsIterable) {
+  console.log(session);
+  if (!allSessionsIterable.isNextCached) {
+    console.log("the next loop is gonna take some time");
+  }
+}
+
+// list sessions associated with a specific room -
+const sessionFilters = {
+  room_id: "test_room_id",
+  limit: 10, // specifies the max no. of objects in one page
+  // this means `iterable.isNextCached` will be `false` once every 10 times
+};
+const sessionsByRoomIterable = hms.sessions.list(sessionFilters);
+for await (const session of sessionsByRoomIterable) {
+  console.log(session);
+}
+```
+
+## Currently Supported Endpoints
+
+1. [Rooms APIs](https://www.100ms.live/docs/server-side/v2/api-reference/Rooms/object)
+2. [Active Rooms APIs](https://www.100ms.live/docs/server-side/v2/api-reference/active-rooms/object)
+3. [Room Codes APIs](https://www.100ms.live/docs/server-side/v2/api-reference/room-codes/room-code-object)
+4. [Sessions](https://www.100ms.live/docs/server-side/v2/api-reference/Sessions/object)
+
+Support for other endpoints are being added. If you want to consume them before it's available in the SDK, feel free to use the `api` property to make the API calls:
+
+```js
+import HMS from "@100mslive/server-sdk";
+
+const hms = new HMS.SDK();
+const hmsObject = await hms.api.get(path, params);
```

Comment:
does this configure the appropriate headers (auth, content-type)? let's add a line on that behaviour.

would be great to have an example of "path" - do i need the full path with the base url? or put `/v2/...`? need example

Reply:
also need example on how to access response body